File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.c
int maxAscendingSum(int* nums, int numsSize) {
    int curr=nums[0],max=nums[0];
    for(int itr =1; itr<numsSize; itr++){
        if(nums[itr-1]<nums[itr]){
            curr+=nums[itr];
        }
        else{
            max=curr>max?curr:max;
            curr=nums[itr];
        }
    }
    return curr>max?curr:max;
}

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int prev=0,max=nums[0],curr=0;
        for(int num:nums){
            if(num>prev){
                curr+=num;
            }
            else{
                max=Math.max(max,curr);
                curr=num;
            }
            prev=num;
        }
        return max>curr?max:curr;
    }
}